### DISCUSSION TITLE
LEI record described using FIBO

### DISCUSSION THREAD

**Discussion by sdml** [edmcouncil/fibo#1361] [Business Entities (BE)]
I've created this RDF graph of LEI record based on data from GLEIF to use as one of export formats on [lei.info](https://lei.info/) site. (_source xml:_ [https://lei.info/J3WHBG0MTS7O8ZVMDC91.xml](https://lei.info/J3WHBG0MTS7O8ZVMDC91.xml)). It looks like this:
```
@prefix fibo-be-corp-corp: <https://spec.edmcouncil.org/fibo/ontology/BE/Corporations/Corporations/> .
@prefix fibo-be-le-fbo: <https://spec.edmcouncil.org/fibo/ontology/BE/LegalEntities/FormalBusinessOrganizations/> .
@prefix fibo-be-le-lei: <https://spec.edmcouncil.org/fibo/ontology/BE/LegalEntities/LEIEntities/> .
@prefix fibo-be-le-lp: <https://spec.edmcouncil.org/fibo/ontology/BE/LegalEntities/LegalPersons/> .
@prefix fibo-fbc-fct-breg: <https://spec.edmcouncil.org/fibo/ontology/FBC/FunctionalEntities/BusinessRegistries/> .
@prefix fibo-fbc-fct-ra: <https://spec.edmcouncil.org/fibo/ontology/FBC/FunctionalEntities/RegistrationAuthorities/> .
@prefix fibo-fbc-fct-usfsind: <https://spec.edmcouncil.org/fibo/ontology/FBC/FunctionalEntities/NorthAmericanEntities/USFinancialServicesEntitiesIndividuals/> .
@prefix fibo-fnd-org-fm: <https://spec.edmcouncil.org/fibo/ontology/FND/Organizations/FormalOrganizations/> .
@prefix fibo-fnd-plc-adr: <https://spec.edmcouncil.org/fibo/ontology/FND/Places/Addresses/> .
@prefix fibo-fnd-plc-loc: <https://spec.edmcouncil.org/fibo/ontology/FND/Places/Locations/> .
@prefix fibo-fnd-rel-rel: <https://spec.edmcouncil.org/fibo/ontology/FND/Relations/Relations/> .
@prefix lcc-3166-1: <https://www.omg.org/spec/LCC/Countries/ISO3166-1-CountryCodes/> .
@prefix lcc-3166-2-us: <https://www.omg.org/spec/LCC/Countries/Regions/ISO3166-2-SubdivisionCodes-US/> .
@prefix lcc-cr: <https://www.omg.org/spec/LCC/Countries/CountryRepresentation/> .
@prefix lcc-lr: <https://www.omg.org/spec/LCC/Languages/LanguageRepresentation/> .
@prefix lei-info: <https://lei.info/> .
@prefix owl: <http://www.w3.org/2002/07/owl#> .
@prefix rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#> .
@prefix rdfs: <http://www.w3.org/2000/01/rdf-schema#> .
@prefix xml: <http://www.w3.org/XML/1998/namespace> .
@prefix xsd: <http://www.w3.org/2001/XMLSchema#> .

<https://lei.info/RegistrationAuthority/RA000625> a fibo-fbc-fct-breg:BusinessRegistry .

<https://lei.info/J3WHBG0MTS7O8ZVMDC91#bei> a fibo-be-corp-corp:RegistrationIdentifier ;
    fibo-fbc-fct-ra:isRegisteredIn <https://lei.info/RegistrationAuthority/RA000625> ;
    fibo-fnd-rel-rel:hasTag "3793794100"^^xsd:string ;
    lcc-lr:identifies lei-info:J3WHBG0MTS7O8ZVMDC91 .

<https://lei.info/J3WHBG0MTS7O8ZVMDC91#entry> a fibo-fbc-fct-breg:LegalEntityIdentifierRegistryEntry ;
    fibo-fbc-fct-breg:hasInitialRegistrationDate "2012-06-06T15:51:00+00:00"^^xsd:dateTime ;
    fibo-fbc-fct-breg:hasRegistrationStatus fibo-fbc-fct-breg:IssuedStatus ;
    fibo-fbc-fct-breg:hasRegistrationStatusRevisionDate "2020-07-24T19:40:00+00:00"^^xsd:dateTime ;
    fibo-fbc-fct-breg:hasRenewalDate "2021-06-17T14:36:00+00:00"^^xsd:dateTime ;
    fibo-fbc-fct-breg:hasValidationLevel fibo-fbc-fct-breg:EntityValidationLevelFullyCorroborated ;
    fibo-fnd-rel-rel:comprises <https://lei.info/J3WHBG0MTS7O8ZVMDC91#lei> ;
    lcc-cr:isPartOf <http://lei.info/EVK05KS7XY1DEII3R011#registry> .

<https://lei.info/RA000625#code> a fibo-fbc-fct-breg:RegistrationAuthorityCode ;
    fibo-fnd-rel-rel:hasTag "RA000625"^^xsd:string ;
    lcc-lr:identifies <https://lei.info/RegistrationAuthority/RA000625> .

<https://lei.info/J3WHBG0MTS7O8ZVMDC91#entityLegalForm> a fibo-be-le-lei:EntityLegalForm ;
    lcc-lr:isIdentifiedBy <https://lei.info/J3WHBG0MTS7O8ZVMDC91#entityLegalFormIdentifier> .

<https://lei.info/J3WHBG0MTS7O8ZVMDC91#entityLegalFormIdentifier> a fibo-be-le-lei:EntityLegalFormIdentifier ;
    fibo-fnd-rel-rel:hasTag "XSNP"^^xsd:string .

<https://lei.info/J3WHBG0MTS7O8ZVMDC91#hqAddress> a fibo-fnd-plc-adr:PhysicalAddress ;
    fibo-fnd-plc-adr:hasAddressLine1 "5959 Las Colinas Boulevard"^^xsd:string ;
    fibo-fnd-plc-adr:hasPostalCode "75039"^^xsd:string ;
    fibo-fnd-plc-loc:hasCity "Irving"^^xsd:string ;
    fibo-fnd-plc-loc:hasCountry lcc-3166-1:UnitedStatesOfAmerica .

<https://lei.info/J3WHBG0MTS7O8ZVMDC91#legalAddress> a fibo-fnd-plc-adr:ConventionalStreetAddress ;
    fibo-fnd-plc-adr:hasAddressLine1 "C/O Corporation Service Company"^^xsd:string ;
    fibo-fnd-plc-adr:hasPostalCode "08628"^^xsd:string ;
    fibo-fnd-plc-loc:hasCity "Ewing"^^xsd:string ;
    fibo-fnd-plc-loc:hasCountry lcc-3166-1:UnitedStatesOfAmerica .

<https://lei.info/J3WHBG0MTS7O8ZVMDC91#otherEntityLegalForm> a fibo-be-le-lei:EntityLegalForm ;
    lcc-lr:hasName "CORPORATION"^^xsd:string .

<http://lei.info/EVK05KS7XY1DEII3R011#registry> a fibo-fbc-fct-breg:LegalEntityIdentifierRegistry .

lei-info:J3WHBG0MTS7O8ZVMDC91 a fibo-be-le-lp:LegalEntity ;
    fibo-be-le-fbo:hasHeadquartersAddress <https://lei.info/J3WHBG0MTS7O8ZVMDC91#hqAddress> ;
    fibo-be-le-lei:hasLegalAddress <https://lei.info/J3WHBG0MTS7O8ZVMDC91#legalAddress> ;
    fibo-be-le-lei:hasLegalForm <https://lei.info/J3WHBG0MTS7O8ZVMDC91#entityLegalForm>,
        <https://lei.info/J3WHBG0MTS7O8ZVMDC91#otherEntityLegalForm> ;
    fibo-fbc-fct-breg:hasEntityStatus fibo-fbc-fct-breg:ActiveStatus ;
    fibo-fbc-fct-breg:hasPriorLegalName "Exxon Corporation"^^xsd:sting ;
    fibo-fnd-rel-rel:hasLegalName "EXXON MOBIL CORPORATION"^^xsd:string ;
    lcc-lr:isIdentifiedBy <https://lei.info/J3WHBG0MTS7O8ZVMDC91#lei> .

<https://lei.info/J3WHBG0MTS7O8ZVMDC91#lei> a fibo-be-le-lei:LegalEntityIdentifier ;
    fibo-fbc-fct-ra:isRegisteredIn <http://lei.info/EVK05KS7XY1DEII3R011#registry> ;
    fibo-fnd-rel-rel:hasTag "J3WHBG0MTS7O8ZVMDC91"^^xsd:string ;
    fibo-fnd-rel-rel:isManagedBy lei-info:EVK05KS7XY1DEII3R011 ;
    lcc-lr:identifies lei-info:J3WHBG0MTS7O8ZVMDC91 .
```

Is it done correctly?
Below is entity diagram of the same data:

![lei-fibo](https://user-images.githubusercontent.com/8113286/101196178-ae105b80-3660-11eb-9176-ea35ab5f7d21.png)

**Comment by ElisaKendall** [comment]
@sdml Hi Dominik, This is really close.  For EntityLegalForm, the identifier should point to the individual legal form, using 'identifies', rather than the other way around.  Also, the relationships between the LEI Registry Entry, the Registry, and the LEI are not quite right.   isRegisteredIn is the correct property from the LEI to the Registry.  Rather than use isPartOf (which is transitive), we've used 'isIncludedIn' for the relationship from the registry entry to the registry.   The other thing is that the registration identifier on the upper left, identifies the legal entity rather than the LEI.  Best regards, Elisa

**Comment by rivettp** [comment]
Some further issues...
I'm not sure about ConventionalStreetAddress which is rather US-specific - why not use PhysicalAddress for both?

Your original representation from lei.info is a bit incomplete for representing all GLEIF info: for example it misses:
- link from the LegalEntity to its Jurisdiction (that's a big one!)
- links from the LEIRegistryEntry to its Validation identifier. 
- link from ELF to its Jurisdiction(s)
- abbreviations for ELF
- various versions of names and addresses (e.g. previous names, transliterated names/addresses).
- geocoded versions of addresses

There is now an official ontology representation from GLEIF, which is available via data.world https://data.world/gleif/lei-data (see also the Getting Started project linked to on the right). You may need a free account. The primary L1 ontology is also documented here https://www.gleif.org/ontology/v1.0/L1/index-en.html 

This data.world project includes a complete data set refreshed weekly and is also available via follow your nose and via a SPARQL SERVICE endpoint or REST interface.
I have nearly completed a formal mapping ontology linking it to FIBO.
We are planning to make the source code available for populating it from the GLEIF-published XML.

One technique worth pointing out at the data level is the use of multiple classification so that the RegistryEntry is also an instance of Identifier for LEI and ELF: separate classes, one individual sicne they're always one-to-one.

Feel free to ping me for further info or comments.

**Comment by sdml** [comment]
Thank you for all your insight. I managed to fix most of mentioned issues.
The thing I can't figure out is what property in FIBO is used to represent LegalJurisdiction of LegalEntiy.

**Comment by rivettp** [comment]
The property is fibo-be-le-cb:isIncorporatedIn or, fully, https://spec.edmcouncil.org/fibo/ontology/BE/LegalEntities/CorporateBodies/isIncorporatedIn
That has range of fibo-fnd-law-jur:Jurisdiction so if you use that property with a country or state it will be inferred to be a Jurisdiction too, which is fine for most uses IMO but if you want to be a purist you'd have a separate individual for the Jurisdiction such as fibo-be-ge-usj:StateOfCaliforniaJurisdiction, which is in FIBO with property linking it to the state fibo-fnd-law-jur:hasReach rdf:resource="&lcc-3166-2-us;California

**Comment by jfgemski** [comment]
FYI – makes no sense to me!

 

John

 

Jfgemski@gmail.com <mailto:Jfgemski@gmail.com>  | M: (516) 353-0004

**Comment by trypuz** [comment]
> FYI – makes no sense to me! 

John, @jfgemski , could you be more precise, please :-) What exactly makes no sense to you?

**Comment by trypuz** [comment]
> Thank you for all your insight. I managed to fix most of mentioned issues.
> The thing I can't figure out is what property in FIBO is used to represent LegalJurisdiction of LegalEntity.

Dominik, @sdml, it seems that Pete, @rivettp , gave you some hints considering modelling LegalJurisdiction. Could you share the final FIBO model for https://lei.info/J3WHBG0MTS7O8ZVMDC91.xml, please?
